From ec1279caa3da11872dd732fb778756c4324b9219 Mon Sep 17 00:00:00 2001
From: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Date: Mon, 11 Nov 2013 16:40:16 -0800
Subject: Dalvik: PCG does not require certain ME passes

BZ: 151253

The PCG backend does not require certain ME passes due to it doing the work
internally.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: I61a6a96fcd90aab837941a7a788ef61593b8423a
Orig-MCG-Change-Id: I8c123f9b48c92b6891eb744ce6d3a2dfd7a33209
Signed-off-by: Jean Christophe Beyler <jean.christophe.beyler@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/PcgInterface.cpp |   15 ++++++++++++++-
 1 files changed, 14 insertions(+), 1 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/PcgInterface.cpp b/vm/compiler/codegen/x86/pcg/PcgInterface.cpp
index 944bd1d..13a49af 100644
--- a/vm/compiler/codegen/x86/pcg/PcgInterface.cpp
+++ b/vm/compiler/codegen/x86/pcg/PcgInterface.cpp
@@ -271,6 +271,17 @@ extern "C" void setupPcgJit (void)
                               (void*)pcgDvmClientGetMemConstSymbol);
 }
 
+/*
+ * @brief Fix the pass driver for PCG
+ */
+static void passHandler (void)
+{
+    //PCG does not need write back information or registerization
+    dvmCompilerRemovePass ("Write_Back_Registers");
+    dvmCompilerRemovePass ("Registerization_ME");
+}
+
+
 /**
  * @brief Plugin initialization
  * @return true if successfully initialized, false if otherwise
@@ -288,8 +299,10 @@ extern "C" bool dalvikPluginInit (void)
     // Initialize PCG.
     pcgModuleBegin ();
 
-    setupPcgJit();
+    setupPcgJit ();
 
+    //Remove ME passes that PCG does not require
+    passHandler ();
 
     return true;
 }
-- 
1.7.4.1

